### DISCUSSION TITLE
Is it possible to add speech recognition (STT/TTS) to a Wails app using a Chrome or Mozilla speech processing API ?

### DISCUSSION THREAD

**Discussion by serge-hulne** [wailsapp/wails#2307] [Q&A]
In theory, the Wails frontend part of a Wails app behaves like a browser, so it should be possible to use a web-based speech processing API with it.

- Has anybody tried that?
- Any pointers/doc/refs on  how to proceed?

I tried using `  var recognition = new webkitSpeechRecognition();`, but `webkitSpeechRecognition` is not recognised as known JS class in  the context of Wails :
 
```
<script lang="ts">
  import {Greet} from '../wailsjs/go/main/App.js'

  let resultText: string = "Please enter your name below 👇"
  let name: string

  let recognition = new webkitSpeechRecognition();

  function greet(): void {
    Greet(name).then(result => resultText = result)
  }

</script>

<main>

  <div class="input-box" id="input">
    <button class="btn" on:click={greet}>Greet</button>
  </div>
</main>

<style>
</style>

```

**Comment by leaanthony** [comment]
The frontend is a webview, not a browser. It is not 100% the same so not all browser APIs are available.

**Reply by serge-hulne** [reply]
Yes, I understand that, but I was wondering if anybody has thought about a workaround?

Perhaps, the answer would lie server-side rather than client-side, perhaps in the shape of a go package?

**Reply by airtonix** [reply]
> the answer would lie server-side rather than client-side, perhaps in the shape of a go package.

Yes.
